packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl

```diff
@@ -4,16 +4,25 @@
 // To add platforms, run `flutter create -t plugin --platforms <platforms> .` under the same
 // directory. You can also find a detailed instruction on how to add platforms in the `pubspec.yaml` at https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms.
 {{/no_platforms}}
-
 import 'dart:async';
 
-import 'package:flutter/services.dart';
+import '{{projectName}}_platform_interface.dart';
 
 class {{pluginDartClass}} {
-  static const MethodChannel _channel = MethodChannel('{{projectName}}');
+  factory {{pluginDartClass}}() {
+    _singleton ??= {{pluginDartClass}}._();
```

Comment:
Why have you restructured this class? You could have kept exactly the same structure and just had the implementation call `{{pluginDartClass}}Platform.instance.platformVersion`, so this doesn't seem related to the goal of the PR.

Reply:
I think I was just bringing it in line with other examples I'd seen for this interface. I'll back it out and it can be a separate PR if desired.